src/Illuminate/Support/Str.php

```diff
@@ -430,15 +430,13 @@ public static function startsWith($haystack, $needles)
      */
     public static function studly($value)
     {
-        $key = $value;
-
-        if (isset(static::$studlyCache[$key])) {
-            return static::$studlyCache[$key];
+        if (isset(static::$studlyCache[$value])) {
+            return static::$studlyCache[$value];
         }
 
         $value = ucwords(str_replace(['-', '_'], ' ', $value));
 
-        return static::$studlyCache[$key] = str_replace(' ', '', $value);
+        return static::$studlyCache[$value] = str_replace(' ', '', $value);
```

Comment:
I'm fairly sure this is a breaking change isnt it? $value is not the same as $key was - because $value was manipulated with a ucwords() function.

i.e. If I had `$value = "someWord"` before.

original code would be `$studyCache['someWord']`

now your code is `$studyCache['Someword']`